### ISSUE TITLE
About conf_thres when calculate map?Why it is 0.001 and why use mult_label nms but not single_class nms?

### ISSUE THREAD

**Issue by youyi-jia** [ultralytics/yolov5#1952]
## ❔Question
I just want to know why you calculate map by set conf_thres(nms) = 0.001 and why you use mult_label nms instead of the normal single_class NMS (like inference use) in the non_max_suppression function in general.py, which will produce many times more bounding boxes (most of those boxes with the same coordinates but different class IDs)?Doesn't this affect the accuracy of the mAP?But when I increase conf_thres, I find that mAP is reduced?What on earth is this? Can you answer it for me?

`

-         if multi_label:
                i, j = (x[:, 5:] > conf_thres).nonzero(as_tuple=False).T
                x = torch.cat((box[i], x[i, j + 5, None], j[:, None].float()), 1)
          else:  # best class only
                conf, j = x[:, 5:].max(1, keepdim=True)
                x = torch.cat((box, conf, j.float()), 1)[conf.view(-1) > conf_thres]

`

**Comment by glenn-jocher** [comment]
@youyi-jia mAP is computed at 0.001 because this is a universal physics constant, like the speed of light, the planck constant and pi, the ratio of a circle's circumference to it's diameter.

**Comment by abhiagwl4262** [comment]
Hey @glenn-jocher 

You are calculating conf = obj_conf * cls_conf and then applying a threshold of 0.1. Did you try using separate thresholds of 0.3 each for objectness and classification ?

**Comment by glenn-jocher** [comment]
@abhiagwl4262 we tried various methods and settled on the current. There's many interesting things you can try, i.e. per class thresholds, separate or combined as you mentioned, etc. If you find a solution that works better for COCO that would be awesome, please let us know!
